src/System.Runtime/tests/Performance/Perf.Double.cs

```diff
@@ -49,6 +50,40 @@ public void DefaultToString(double number, int innerIterations)
             }
         }
 
+        [Benchmark]
+        [InlineData("-∞", 10_000_000)]                      // Negative Infinity
+        [InlineData("-1.7976931348623157E+308", 100_000)]   // Min Negative Normal
+        [InlineData("-3.1415926535897931", 1_000_000)]      // Negative pi
```

Comment:
How about numbers with commas (thousands separator)?

Reply:
Also - numbers with white space at the beginning/end.  from [https://docs.microsoft.com/en-us/dotnet/api/system.double.tryparse](https://docs.microsoft.com/en-us/dotnet/api/system.double.tryparse)

```
or a string of the form:

[ws][sign][integral-digits,]integral-digits[.[fractional-digits]][e[sign]exponential-digits][ws]
```

---
In reply to: [219635010](https://github.com/dotnet/corefx/pull/32392#discussion_r219635010) [](ancestors = 219635010)

Reply:
These few tests are already adding on 6min 20sec onto the total benchmark time (38 new tests at 10s each). I think there are a number of other scenarios that would be good to test, but the current Benchmark.NET setup does not make that very feasible, so I tried to cover a number of useful/interesting inputs for the most common string format.

Reply:
> so I tried to cover a number of useful/interesting inputs for the most common string format.

From a cursory look, I think some of your existing ones can be eliminated.  For example:

```
System.Tests.Perf_Single.DefaultTryParse(input: "2.71828175", innerIterations: 1000000)                                                          | Duration | msec |    116     |     86.756 |      3.454 |     82.339 |  104.550
System.Tests.Perf_Single.DefaultTryParse(input: "-2.71828175", innerIterations: 1000000)                                                         | Duration | msec |    106     |     94.820 |      8.288 |     85.799 |  131.644
System.Tests.Perf_Single.DefaultTryParse(input: "3.14159274", innerIterations: 1000000)                                                          | Duration | msec |    108     |     92.847 |      7.307 |     83.224 |  120.448
System.Tests.Perf_Single.DefaultTryParse(input: "-3.14159274", innerIterations: 1000000)                                                         | Duration | msec |    115     |     87.533 |      3.175 |     84.733 |  106.748
```

Yes, these are important numbers: `e` and `pi`, but is parsing these numbers really that much different of a scenario?

Similar comment for:

```
   System.Tests.Perf_Double.DefaultTryParse(input: "1.7976931348623157E+308", innerIterations: 100000)                                              | Duration | msec |    863     |     11.594 |      0.479 |     11.033 |   15.891
   System.Tests.Perf_Double.DefaultTryParse(input: "-1.7976931348623157E+308", innerIterations: 100000)                                             | Duration | msec |    776     |     12.890 |      1.045 |     11.703 |   22.712
   System.Tests.Perf_Double.DefaultTryParse(input: "2.2250738585072009E-308", innerIterations: 100000)                                              | Duration | msec |    805     |     12.421 |      0.531 |     11.805 |   18.015
   System.Tests.Perf_Double.DefaultTryParse(input: "-2.2250738585072009E-308", innerIterations: 100000)                                             | Duration | msec |    777     |     12.867 |      0.841 |     12.156 |   24.272
   System.Tests.Perf_Double.DefaultTryParse(input: "2.2250738585072014E-308", innerIterations: 100000)                                              | Duration | msec |    845     |     11.839 |      0.536 |     11.271 |   18.168
   System.Tests.Perf_Double.DefaultTryParse(input: "-2.2250738585072014E-308", innerIterations: 100000)                                             | Duration | msec |    827     |     12.096 |      0.389 |     11.574 |   15.560
```

Reply:
Talked with @eerhardt briefly offline.

I definitely agree that we should add more tests, but I am waiting to hear back on if we have or plan to have a better story for "end to end" benchmarks (so that we can actually validate the average perf for a broad range of inputs, or for benchmarks that take longer than 1 second per outer iteration to run).

The current tests where chosen because they have the broadest impact on the current algorithm (overall) and will be the most useful in validating some related changes going in.